catalogd/internal/serverutil/serverutil_test.go

```diff
@@ -0,0 +1,254 @@
+		"replaces": "cockroachdb.v5.0.3"
+	  }
+	],
+	"name": "stable-5.x",
+	"package": "cockroachdb",
+	"schema": "olm.channel"
+  }
+  {
+	"entries": [
+	  {
+		"name": "cockroachdb.v6.0.0",
+		"skipRange": "<6.0.0"
+	  }
+	],
+	"name": "stable-v6.x",
+	"package": "cockroachdb",
+	"schema": "olm.channel"
+  }
+  {
+	"image": "quay.io/openshift-community-operators/cockroachdb@sha256:a5d4f4467250074216eb1ba1c36e06a3ab797d81c431427fc2aca97ecaf4e9d8",
+	"name": "cockroachdb.v5.0.3",
+	"package": "cockroachdb",
+	"properties": [
+	  {
+		"type": "olm.gvk",
+		"value": {
+		  "group": "charts.operatorhub.io",
+		  "kind": "Cockroachdb",
+		  "version": "v1alpha1"
+		}
+	  },
+	  {
+		"type": "olm.package",
+		"value": {
+		  "packageName": "cockroachdb",
+		  "version": "5.0.3"
+		}
+	  }
+	],
+	"relatedImages": [
+	  {
+		"name": "",
+		"image": "quay.io/helmoperators/cockroachdb:v5.0.3"
+	  },
+	  {
+		"name": "",
+		"image": "quay.io/openshift-community-operators/cockroachdb@sha256:a5d4f4467250074216eb1ba1c36e06a3ab797d81c431427fc2aca97ecaf4e9d8"
+	  }
+	],
+	"schema": "olm.bundle"
+  }
+  {
+	"image": "quay.io/openshift-community-operators/cockroachdb@sha256:f42337e7b85a46d83c94694638e2312e10ca16a03542399a65ba783c94a32b63",
+	"name": "cockroachdb.v5.0.4",
+	"package": "cockroachdb",
+	"properties": [
+	  {
+		"type": "olm.gvk",
+		"value": {
+		  "group": "charts.operatorhub.io",
+		  "kind": "Cockroachdb",
+		  "version": "v1alpha1"
+		}
+	  },
+	  {
+		"type": "olm.package",
+		"value": {
+		  "packageName": "cockroachdb",
+		  "version": "5.0.4"
+		}
+	  }
+	],
+	"relatedImages": [
+	  {
+		"name": "",
+		"image": "quay.io/helmoperators/cockroachdb:v5.0.4"
+	  },
+	  {
+		"name": "",
+		"image": "quay.io/openshift-community-operators/cockroachdb@sha256:f42337e7b85a46d83c94694638e2312e10ca16a03542399a65ba783c94a32b63"
+	  }
+	],
+	"schema": "olm.bundle"
+  }
+  {
+	"image": "quay.io/openshift-community-operators/cockroachdb@sha256:d3016b1507515fc7712f9c47fd9082baf9ccb070aaab58ed0ef6e5abdedde8ba",
+	"name": "cockroachdb.v6.0.0",
+	"package": "cockroachdb",
+	"properties": [
+	  {
+		"type": "olm.gvk",
+		"value": {
+		  "group": "charts.operatorhub.io",
+		  "kind": "Cockroachdb",
+		  "version": "v1alpha1"
+		}
+	  },
+	  {
+		"type": "olm.package",
+		"value": {
+		  "packageName": "cockroachdb",
+		  "version": "6.0.0"
+		}
+	  }
+	],
+	"relatedImages": [
+	  {
+		"name": "",
+		"image": "quay.io/cockroachdb/cockroach-helm-operator:6.0.0"
+	  },
+	  {
+		"name": "",
+		"image": "quay.io/openshift-community-operators/cockroachdb@sha256:d3016b1507515fc7712f9c47fd9082baf9ccb070aaab58ed0ef6e5abdedde8ba"
+	  }
+	],
+	"schema": "olm.bundle"
+  }
+  `
+
+// mockStorageInstance implements storage.Instance interface for testing
+type mockStorageInstance struct {
+	content string
+}
+
+func (m *mockStorageInstance) StorageServerHandler() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte(m.content))
```

Comment:
Do we actually care what this content is, or do we just care about its length? If the latter, I'd suggest removing the `testCompressableJSON` variable and replacing with with something like `strings.Rand(sizeThreshold+1)`